ui/src/app/registration/modal/modal.component.ts

```diff
@@ -96,30 +102,31 @@ export class RegistrationModalComponent implements OnInit {
   private getForm(role: string): FormGroup {
     if (role === 'installer') {
       return this.formBuilder.group({
-        companyName: new FormControl("", Validators.required),
-        firstname: new FormControl("", Validators.required),
-        lastname: new FormControl("", Validators.required),
-        street: new FormControl("", Validators.required),
-        zip: new FormControl("", Validators.required),
-        city: new FormControl("", Validators.required),
-        country: new FormControl("", Validators.required),
-        phone: new FormControl("", Validators.required),
-        email: new FormControl("", [Validators.required, Validators.email]),
-        password: new FormControl("", Validators.required),
-        confirmPassword: new FormControl("", Validators.required),
+        companyName: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        firstname: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        lastname: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        street: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        zip: new FormControl("", [Validators.required, Validators.pattern('^[0-9]{5}$')]),
+        city: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        country: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        phone: new FormControl("", [Validators.required, Validators.pattern('^[0-9]+$')]),
+        email: new FormControl("", [Validators.required, emailValidator, Validators.maxLength(50)]),
+        password: new FormControl("", [Validators.required, Validators.minLength(8), Validators.maxLength(20)]),
+        confirmPassword: new FormControl("", [Validators.required, Validators.minLength(8), Validators.maxLength(20)]),
+        Installateur: new FormControl("", [Validators.required, Validators.maxLength(50)]),
```

Comment:
It's not part of your Pull Request, but this should actually be named `installer`...